samples/azureeventsourcelistener-eventhubs-logging/Program.cs

```diff
@@ -0,0 +1,140 @@
+                instrumentationKey = args[2];
+            }
+
+            // If there was no connection string provided, then prompt for one.
+            while (string.IsNullOrEmpty(connectionString))
+            {
+                Console.Write("Please provide the connection string for the Event Hubs that you'd like to use and then press Enter: ");
+                connectionString = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            // If there was no eventHubName provided, then prompt for one.
+            while (string.IsNullOrEmpty(eventHubName))
+            {
+                Console.Write("Please provide the Event Hub Name that you'd like to use and then press Enter: ");
+                eventHubName = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            // If there was no instrumentation Key provided, then prompt for one.
+            while (string.IsNullOrEmpty(instrumentationKey))
+            {
+                Console.Write("Please provide the Event Hub Name that you'd like to use and then press Enter: ");
+                instrumentationKey = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            //Create ApplicationInsights Client.
+            var telemetryClient = new TelemetryClient(new TelemetryConfiguration(instrumentationKey));
+            
+            //Create EventHub Producer Client.
+            var producerClient = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                // Setup a listener to monitor logged events.
+                var listener = new AzureEventSourceListener((args, message) =>
+                {
+                    // Add condition to manipulate only events related to EventHubs.
+                    if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+                    {
+                        var properties = GetPropertiesDictionnary(args.PayloadNames, args.Payload);
+                        var level = args.Level.GetSeverityLevel();
+                        var traceMessage = string.Format(args.Message, args.Payload.ToArray());
+                        telemetryClient.TrackTrace($"Sample_AzureEventSourceListener | {traceMessage}", level, properties);
+                        telemetryClient.Flush();
+                    }
+                }, EventLevel.LogAlways);
+
+                // Creating a producer client using its default set of options.
+                // We will publish three small batch events.
+
+                Console.WriteLine("Preparing event batch...");
+                using EventDataBatch eventBatch = await producerClient.CreateBatchAsync();
+                // Add events to the batch. An event is a represented by a collection of bytes and metadata.
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Hello, Event Hubs!")));
```

Comment:
We found that our previous samples caused confusion about best practices when we didn't  check the return value of `TryAdd`.    Can we shift the pattern to checking and throwing if `TryAdd` returns `false`?   An example can be found [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishingEvents.md#publishing-size-constraints).